FACULTY_BRANCH_PROTECTION_SETTINGS.md

```diff
@@ -0,0 +1,55 @@
+# Branch protection settings for this lab <!-- omit in toc -->
+
+* [Preparing for the first part (before the lab starts)](#preparing-for-the-first-part-before-the-lab-starts)
+* [Switching things for the second part](#switching-things-for-the-second-part)
+
+> :mage_woman: This documentation is primarily for the faculty (and
+> possibly TAs) running this lab. If it's useful for students, though,
+> that's cool.
+
+There are two main parts to this lab:
+
+* The first part, where students just commit new, separate Markdown
+  files.
+* The second part, where students commit changes to a shared Java file.
+
+## Preparing for the first part (before the lab starts)
+
+In the first part the students commit directly to `main`, and that part
+is designed to make it very unlikely that they will have any merge
+conflicts. There are also no substantial "checks" that apply to these
+Markdown files.
+
+Because they'll be committing directly to `main`, and there are no
+meaningful checks, it's important that all branch protections are
+_turned off_ (on the copy created by GitHub Classroom in that semester's
+organization) before they start the lab.
+
+To check this, go to the GitHub web interface for the (copied) repository
+and go to "Settings > Branches". If there's no branch protection rule
+for `main` then everything's fine. If there is, however, then you want to
+edit it, uncheck _everything_, and then click the green "Save changes"
+button at the bottom. (You could also just delete the branch protection,
+but then you'd have to recreate it when you're starting the second part.)
```

Comment:
There are some notable differences between not having a branch protection rule for a branch and having one with nothing selected:

> By default, each branch protection rule disables force pushes to the matching branches and prevents the matching branches from being deleted

(from https://docs.github.com/en/github/administering-a-repository/about-protected-branches)

Reply:
Good catch. So having one with nothing selected would prevent people from pushing in the first half of the lab, which would be Very Bad. So we really need folks to make sure there is _no_ branch protection.

Reply:
It would still let them push but restricts force pushes which is actually a good thing I think.

Reply:
Oh, I this I misunderstood. So we _do_ want/prefer an "empty" (all unchecked) protection rather than no protection at all?